### PR TITLE
kernel: Add GlobalSec.

### PR DESCRIPTION
**Pull request by osenft**
This adds explicit support for H1's Global Security Module,
in particular to automatically configure flash protection
for the inactive RO and RW.

### CODE REVIEW

File: kernel/h1/src/globalsec.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+impl GlobalSecHardware {
+    const fn new(base_addr: StaticRef<Registers>) -> GlobalSecHardware {
+        let globalsec = GlobalSecHardware {
+            registers: base_addr,
+            runtime_segment_info: UNKNOWN_RUNTIME_SEGMENT_INFO,
+        };
+
+        globalsec
```
Comment (jrvanwhy) [inline kernel/h1/src/globalsec.rs]: You don't need the `globalsec` local variable here, just return the new struct directly.

File: kernel/h1_syscalls/src/globalsec.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    fn command(&self, command_num: usize, _arg1: usize, _arg2: usize, caller_id: AppId)
+        -> ReturnCode {
+        if self.current_user.get() == None {
+            self.current_user.set(Some(caller_id));
+        }
+        match command_num {
+            0 /* Check if present */ => ReturnCode::SUCCESS,
+            1 /* Get runtime segmenet info */ => self.get_runtime_segment_info(caller_id),
```
Comment (jrvanwhy) [inline kernel/h1_syscalls/src/globalsec.rs]: segment is misspelled in the comment.

File: kernel/h1_syscalls/src/globalsec.rs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        match minor_num {
+            0 => {
+                // Buffer for data exchange
+                self.apps
+                    .enter(app_id, |app_data, _| {
+                        if let Some(s) = slice {
+                            app_data.buffer = Some(s);
+                        } else {
+                            app_data.buffer = slice;
+                        }
```
Comment (jrvanwhy) [inline kernel/h1_syscalls/src/globalsec.rs]: Isn't this whole `if` statement just `app_data.buffer = slice;`?

File: kernel/h1/src/globalsec.rs (hunk 1 above)
Comment (osenft) [reply kernel/h1/src/globalsec.rs]: No idea what I was thinking there. Fixed.

File: kernel/h1_syscalls/src/globalsec.rs (hunk 3 above)
Comment (osenft) [reply kernel/h1_syscalls/src/globalsec.rs]: I copied this pattern a few capsules ago from somewhere ... I think. Anyway. Good point. Fixed.

File: kernel/h1_syscalls/src/globalsec.rs (hunk 2 above)
Comment (osenft) [reply kernel/h1_syscalls/src/globalsec.rs]: Argh. I kept making this same typo and though I had caught it everywhere. Thanks. Fixed.

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**